hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java

```diff
@@ -487,23 +490,23 @@ public void close() throws IOException {
     }
     closed = true;
     try {
-      // If stripe buffer is not empty, encode and flush the stripe.
-      if (ecChunkBufferCache.getFirstDataCell().position() > 0) {
-        generateParityCells();
-        addStripeToQueue(ecChunkBufferCache);
-      }
-      // Send EOF mark to flush thread.
-      addStripeToQueue(new EOFDummyStripe());
+      if (!closing) {
```

Comment:
I wonder if we should still at least try to flush the "stripeQueue" even if the state is closing = true?

When we do writes, the stripe gets buffered in the queue, so there might be some buffered data ready to write out. However closing = true is an error state, indicating something unexpected happened during a previous write, so the completeness of the current stripe buffer may not be known or good.

I am not 100% sure what the best approach is - trying to write out the buffered data could run into problems too, due to the same exception that caused the stream to go into the closing state.